src/app/features/registration/registration.component.ts

```diff
@@ -55,40 +53,33 @@ export class RegistrationComponent extends HelperInputPassword implements OnDest
     if (!this.form.valid) {
       this.form.markAllAsTouched();
     } else {
-      this._authService
-        .register(this.form.value)
-        .pipe(
-          takeUntil(this.destroy$),
-          finalize(() => this.openModal())
-        )
-        .subscribe({
-          next: () => {
-            this.isRegistered = true;
-            this.optionModalInfo.message = 'Registration successful!';
-          },
-          error: () => {
-            this.isRegistered = false;
-            this.optionModalInfo.message = 'A user with this email already exists. Please login or use a different email';
-          },
-        });
+      this.authStateFacade.register(this.form.value);
+      this.authStateFacade.isLoading$.pipe(takeUntil(this.destroy$)).subscribe(isLoading => {
+        if (!isLoading) {
+          this.openModal();
+        }
+      });
     }
   }
 
-  confirmButtonModal(): void {
-    if (this.isRegistered) {
-      this._router.navigateByUrl('/login');
-    }
-    this.closeModal();
-  }
-
   openModal(): void {
+    this.optionModalInfo.message = this.isRegistered
+      ? 'Registration successful!'
```

Comment:
It's better to use `enum` or `const` for string value.

Reply:
move to enum